File: v1/src/main/java/com/api/v1/mappers/CarResponseMapper.java
package com.api.v1.mappers;

import com.api.v1.domain.Car;
import com.api.v1.dtos.CarResponseDto;

public record CarResponseMapper(Car car) {

    public CarResponseDto mapToDto() {
        return new CarResponseDto(
                car.getMake(),
                car.getModel(),
                car.getProductionYear(),
                car.getVin(),
                car.getPlateNumber(),
                car.getAddedAt(),
                car.getUpdatedAt()
        );
    }

}

File: v2/src/main/kotlin/com/api/v2/car/mappers/CarResponseMapper.kt
package com.api.v2.car.mappers

import com.api.v2.car.domain.Car
import com.api.v2.car.dto.CarResponseDto

class CarResponseMapper {

    companion object {
        fun mapToDto(car: Car): CarResponseDto {
            return CarResponseDto(
                car.vin,
                car.model,
                car.productionYear,
                car.make,
                car.plateNumber,
                car.createdAt
            )
        }
    }

}